### PR TITLE
ENH: simplify and optimize some proximal operators

### PR DESCRIPTION
**Pull request by kohr-h**

### CODE REVIEW

File: odl/solvers/nonsmooth/proximal_operators.py [hunk 1]
```diff
@@ -1142,10 +1154,68 @@ def proximal_l1(space, lam=1, g=None, isotropic=False):
...
+            if g is not None:
+                diff = x - g
+            else:
+                if x is out:
+                    # Handle aliased data properly
+                    diff = x.copy()
+                else:
+                    diff = x
+
+            if isotropic:
```
Comment (adler-j) [inline odl/solvers/nonsmooth/proximal_operators.py]: The isotropic case should be moved to another function imo, it is somewhat fundamentally different.

File: odl/solvers/nonsmooth/proximal_operators.py [hunk 2]
```diff
@@ -1251,33 +1321,29 @@ def __init__(self, sigma):
...
             """Apply the operator to ``x`` and stores the result in ``out``."""
+            # (x + lam - sqrt((x - lam)^2 + 4*lam*sig*g)) / 2
 
-            # 1 / 2 (lam_X + x - sqrt((x - lam_X) ^ 2 + 4; lam sigma g)
-
-            # out = x - lam_X
-            out.assign(x)
+            # out = (x - lam)^2
+            if x is out:
+                x = x.copy()
```
Comment (adler-j) [inline odl/solvers/nonsmooth/proximal_operators.py]: Mention why in a comment

**Comment by kohr-h** [comment]
Done, merging after CI

**Comment by kohr-h** [comment]
> Done, merging after CI

No, I'll do the same for the convex conjugate first.

**Comment by kohr-h** [comment]
Now this can be merged (unless CI complains).

**Pull request merged**